rocketpy/tools.py

```diff
@@ -23,3 +26,978 @@ def __get__(self, instance, owner=None):
+
+    def __sub__(self, other):
+        """Subtract two 3x3 matrices."""
+        return Matrix(
+            [
+                [self.xx - other[0][0], self.xy - other[0][1], self.xz - other[0][2]],
+                [self.yx - other[1][0], self.yy - other[1][1], self.yz - other[1][2]],
+                [self.zx - other[2][0], self.zy - other[2][1], self.zz - other[2][2]],
+            ]
+        )
+
+    def __rsub__(self, other):
+        """Subtract two 3x3 matrices."""
+        return Matrix(
+            [
+                [other[0][0] - self.xx, other[0][1] - self.xy, other[0][2] - self.xz],
+                [other[1][0] - self.yx, other[1][1] - self.yy, other[1][2] - self.yz],
+                [other[2][0] - self.zx, other[2][1] - self.zy, other[2][2] - self.zz],
+            ]
+        )
+
+    def __mul__(self, other):
+        """Element wise multiplication of 3x3 matrix self by scalar other."""
+        return Matrix(
+            [
+                [other * self.xx, other * self.xy, other * self.xz],
+                [other * self.yx, other * self.yy, other * self.yz],
+                [other * self.zx, other * self.zy, other * self.zz],
+            ]
+        )
+
+    def __rmul__(self, other):
+        """Element wise multiplication of 3x3 matrix self by scalar other."""
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        """Element wise division is carried out."""
+        return Matrix(
+            [
+                [self.xx / other, self.xy / other, self.xz / other],
+                [self.yx / other, self.yy / other, self.yz / other],
+                [self.zx / other, self.zy / other, self.zz / other],
+            ]
+        )
+
+    def __matmul__(self, other):
+        """Dot (inner) product between two 3x3 matrices or between 3x3 matrix
+        and R3 vector.
+
+        Parameters
+        ----------
+        other : Matrix or Vector
+            The other matrix or vector.
+
+        Returns
+        -------
+        Matrix or Vector
+            The result of the dot product. A Matrix if other if Matrix, and
+            a Vector if other is Vector.
+
+        Examples
+        --------
+        >>> M = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        >>> v = Vector([1, 2, 3])
+        >>> M @ v
+        Vector(14, 32, 50)
+
+        >>> M = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        >>> N = Matrix([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        >>> M @ N
+        Matrix([30, 36, 42],
+               [66, 81, 96],
+               [102, 126, 150])
+        """
+        try:
+            result = [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+            for i, j in product(range(3), range(3)):
+                result[i][j] = Vector(self[i, :]) @ Vector(
+                    [other[0][j], other[1][j], other[2][j]]
+                )
+            return Matrix(result)
+        except (TypeError, IndexError):
+            return Vector([Vector(self[i, :]) @ other for i in range(3)])
```

Comment:
I will test the difference in speed later. However, the only way I can think of speeding up this method is an explicit computation in a 3x3 format. I think it is better to handle the `other` type with an `len` comparison.

Reply:
`len` of both Matrices and Vectors is `3` unfortunately. But I did implement the explicit 3x3 computation to speed up execution significantly. Furthermore, I tried the Laderman algorithm, but it is much slower in Python.